### PR TITLE
Add jax.scipy.linalg.helmert

### PR DESCRIPTION
**Pull request by Saumay**
Adds `jax.scipy.linalg.helmert`, continuing the special-matrix series from #10144 (after `hadamard`, `circulant`, `dft`, `companion` in #37333, `fiedler` in #37334, and `leslie` in #37335).

For row index `k = 1, ..., n - 1`, the contrast row has `k` entries of `1/sqrt(k(k+1))`, then `-k/sqrt(k(k+1))`, then zeros. The implementation just builds those rows from broadcast `arange` indices and two `jnp.where` masks. With `full=True`, a row of `1/sqrt(n)` is prepended so the result is orthogonal.

Static `n` and `full` via `jit`, same pattern as `hadamard` and `hilbert`. Raises `ValueError` for `n < 1`. The `n = 1` corner case still works: returns `(0, 1)` for `full=False` and `[[1.]]` for `full=True`, matching SciPy.

One subtlety worth flagging: the integer indices need an explicit cast to the default float dtype before division, otherwise strict-promotion mode rejects mixing `int32` with `float32` in `1.0 / denom` and `-k / denom`. I caught that the first time tests ran. Casting upfront with `k_f = k.astype(dtypes.default_float_dtype())` fixes it cleanly, and the inner `where` uses `jnp.zeros_like(denom)` instead of `0.0` for the same reason.

### Tests
`testHelmert` runs parity vs SciPy across `n` ∈ {1, 2, 3, 4, 5, 8} × `full` ∈ {False, True}, giving 13 cases under both x32 and x64. `testHelmertInvalidN` covers `n ∈ {0, -1, -5}`. Doctest passes. I also sanity-checked locally that `H @ H.T = I` holds for `full=True` up to `n = 32`, and that contrast rows sum to zero (the property that makes Helmert useful for compositional analysis).

Issue: #10144

### CODE REVIEW

File: jax/_src/scipy/linalg.py [hunk 1]
```diff
@@ -2706,6 +2706,46 @@ def _binom(n, k):
   return lax.exp(a - b - c)
 
 
+@partial(jit, static_argnames=("n", "full"))
```
Comment (jakevdp) [inline jax/_src/scipy/linalg.py]: Please avoid `partial` with `jit`; use `@jit(static_argnames=("n", "full"))`.

**Review by jakevdp — CHANGES_REQUESTED** [review]
One small change. Also please rebase against the updated main branch to fix conflicts. Thanks!

File: jax/_src/scipy/linalg.py (hunk 1 above)
Comment (Saumay) [reply jax/_src/scipy/linalg.py]: Sounds good, switched to `@jit(static_argnames=("n", "full"))` and rebased on main in 7f1958613.

**Review by jakevdp — APPROVED** [review]
Thanks!
